### ISSUE TITLE
Spotify crashes on run with Fedora XFCE spin

### ISSUE THREAD

**Issue by jsm3031** [flathub/com.spotify.Client#75]
Hi, I was using Spotify flatpak fine on Fedora KDE spin. I setup an 2010 13" MBP yesterday with Fedora XFCE spin.

flatpak run com.spotify.Client fails to run and outputs the following repeatedly:

[0308/100041.534245:ERROR:gles2_cmd_decoder.cc(2664)] [GroupMarkerNotSet(crbug.com/242999)!:9032A00300000000]GL ERROR :GL_OUT_OF_MEMORY : BackFramebuffer::Create: <- error from previous GL command
[0308/100041.820039:ERROR:buffer_manager.cc(491)] [.BrowserCompositor-0x45c4fb0]GL ERROR :GL_OUT_OF_MEMORY : glBufferData: <- error from previous GL command
[0308/100041.820382:ERROR:gles2_cmd_decoder.cc(5699)] Error: 5 for Command kBufferData
[0308/100041.820444:ERROR:gles2_cmd_decoder.cc(4570)]   GLES2DecoderImpl: Trying to make lost context current.
[0308/100041.861728:ERROR:gles2_cmd_decoder.cc(2664)] [GroupMarkerNotSet(crbug.com/242999)!:30E4B30300000000]GL ERROR :GL_OUT_OF_MEMORY : BackFramebuffer::Create: <- error from previous GL command
[0308/100041.890960:ERROR:buffer_manager.cc(491)] [.BrowserCompositor-0x45d1c30]GL ERROR :GL_OUT_OF_MEMORY : glBufferData: <- error from previous GL command
[0308/100041.891052:ERROR:gles2_cmd_decoder.cc(5699)] Error: 5 for Command kBufferData
[0308/100041.891136:ERROR:gles2_cmd_decoder.cc(4570)]   GLES2DecoderImpl: Trying to make lost context current.

Thanks for your help

**Comment by jsm3031** [comment]
Fedora 29 btw

**Comment by TingPing** [comment]
I doubt this has anything to do with this package unless you can't reproduce it outside of Flatpak.

**Comment by jsm3031** [comment]
Hi and thanks for your answer!

Could you please elaborate on how I'd do that?  I'm using Flatpak on Fedora and as far as I know Spotify only provides a deb of it or a snap. I have no clue how the Flathub maintainers even get the binary.

How would I run it outside of Flatpak? 

I did assume that the Xfce spin was more slimmed down then the KDE one and maybe I just needed to install some packages that the Flatpak assumes are there.  Could you please advise how I'd proceed with your suggestion?

**Comment by TingPing** [comment]
> maybe I just needed to install some packages that the Flatpak assumes are there.

That isn't how Flatpak works, it doesn't use host libraries.

> Could you please elaborate on how I'd do that? I'm using Flatpak on Fedora and as far as I know Spotify only provides a deb of it or a snap. I have no clue how the Flathub maintainers even get the binary.

Well the Flatpak just extracts the .deb... I guess you can try that on the host.

**Comment by jsm3031** [comment]
I am sorry, I must not be experienced enough to try your suggestions.

I thought I understood that flatpaks use container (or at least container-like) technology/techniques, which I understand to mean that all the dependencies are packaged with the distributed app. When I mentioned the host, I was thinking that can be the only place a difference could exist, but this makes no sense still, since they are both Fedora.

I also thought I understood that Flatpak provides stable platforms: runtimes provide platforms of common libraries that you can depend on, as well as, consistent environments: develop and test your application in an environment that’s identical to the one users have.

So for me, I really cannot understand why it would work on Fedora KDE but not Fedora Xfce.

The other thing I do not understand is how I could test Spotify on my Fedora host outside of the Flatpak since I am sure that is not a way that any sane person would try to run Spotify ;)

At any rate, I tried installing via snap and it works, so thanks for your time!

PS It just occurred to me that running it via snap counts as "can't reproduce it outside of Flatpak". Since I'd rather use Flatpak than snap, if you want to trouble shoot to get this Flatpak working on Fedora Xfce, I'd be willing to continue. Otherwise you may close the ticket. Thanks again for your time.

**Comment by jsm3031** [comment]
So after some more looking into the flatpak/flathub projects and how they work, I must say thanks for all your work! 

After playing around with various things, I ended up realizing that glxgears was broken on this system also and realized it was something to do with this ancient graphics card. Since the system is new I'd risk breaking it and try the Nvidia drivers out.

And now the Flathub Spotify is working. So yes it was the graphics card. Still not sure why the snap works. I should mention that during this playing around I learned that the snap Zoom would not work, even though the Flathub Zoom does...

Thanks again for your time and all the work in this project.
